### PR TITLE
Add SPI for AI Provider

### PR DESCRIPTION
**Pull request by amanmishra22**
Exposed Interface for using AI services which currently has summarizeApp method which will be used to summarize the application in specified format.

### CODE REVIEW

File: cdap-ml-spi/pom.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <packaging>jar</packaging>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-proto</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+  </dependencies>
+</project>
```
Comment (vsethi09) [inline cdap-ml-spi/pom.xml]: Add new line at the end of the file.

File: cdap-ml-spi/pom.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>io.cdap.cdap</groupId>
+    <artifactId>cdap</artifactId>
+    <version>6.11.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>cdap-ml-spi</artifactId>
```
Comment (vsethi09) [inline cdap-ml-spi/pom.xml:14]: Add this module in cdap/pom.xml

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+package io.cdap.cdap.ml.spi;
+
+import io.cdap.cdap.proto.ApplicationDetail;
+
+/**
+ * Summarizes the given application details in mentioned format.
```
Comment (vsethi09) [inline cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Fix and generalize the comment.

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (vsethi09) [inline cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Add copyright header.

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+package io.cdap.cdap.ml.spi;
```
Comment (vsethi09) [inline cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Create subfolders for this package.

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+import io.cdap.cdap.proto.ApplicationDetail;
+
+/**
+ * Summarizes the given application details in mentioned format.
+ */
+public interface AIService {
+
+     /**
+      *
```
Comment (vsethi09) [inline cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Add the description for the function.

**Review by vsethi09 — COMMENTED** [review]
Please address the comments.

File: cdap-ml-spi/pom.xml (hunk 1 above)
Comment (amanmishra22) [reply cdap-ml-spi/pom.xml]: Done

File: cdap-ml-spi/pom.xml (hunk 2 above)
Comment (amanmishra22) [reply cdap-ml-spi/pom.xml:14]: Done

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java (hunk 3 above)
Comment (amanmishra22) [reply cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Done

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java (hunk 6 above)
Comment (amanmishra22) [reply cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Done

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java (hunk 5 above)
Comment (amanmishra22) [reply cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Done

File: cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java (hunk 4 above)
Comment (amanmishra22) [reply cdap-ml-spi/src/main/java/io.cdap.cdap.ml.spi/AIService.java]: Done

**Review by vsethi09 — APPROVED** [review]

**Pull request merged**